intermediate_source/inductor_debug_cpu.py

```diff
@@ -0,0 +1,600 @@
+#
+# 	torch._inductor.debug: [WARNING] model___20 debug trace: /tmp/torchinductor_root/rx/crxfi2ybd7yp5sbj2pnhw33wfhtdw7wumvrobyp5sjvdui5ktjc2.debug
+#
+# In this directory, the following files are saved for debugging purposes:
+#
+# +-------------------------+----------------------------------------------------------+
+# | File                    | Description                                              |
+# +=========================+==========================================================+
+# | fx_graph_runnable.py    | Executable FX graph, post decomps, pre pattern match     |
+# +-------------------------+----------------------------------------------------------+
+# | fx_graph_transformed.py | Transformed FX graph, post pattern match                 |
+# +-------------------------+----------------------------------------------------------+
+# | ir_post_fusion.txt      | Inductor IR before fusion                                |
+# +-------------------------+----------------------------------------------------------+
+# | ir_pre_fusion.txt       | Inductor IR after fusion                                 |
+# +-------------------------+----------------------------------------------------------+
+# | output_code.py          | Generated Python code for graph, with C++/triton kernels |
+# +-------------------------+----------------------------------------------------------+
+#
+# Note that ``fx_graph_runnable.py`` and ``output_code.py`` are both runnable and editable in order to make debugging easier. 
+# Here are the main parts of code extracted from the files:
+#
+# Fx_graph_runnable:
+
+def forward(self, arg0_1, arg1_1):
+    neg = torch.ops.aten.neg.default(arg0_1);  arg0_1 = None
+    maximum = torch.ops.aten.maximum.default(arg1_1, neg);  arg1_1 = neg = None
+    clone = torch.ops.aten.clone.default(maximum);  maximum = None
+    return (clone,)
+
+######################################################################
+# C++ kernel in output_code:
+
+from torch._inductor.codecache import AsyncCompile
+async_compile = AsyncCompile()
+
+cpp_fused_cat_maximum_neg_0 = async_compile.cpp('''
+#include "/tmp/torchinductor_root/gv/cgv6n5aotqjo5w4vknjibhengeycuattfto532hkxpozszcgxr3x.h"
+extern "C" void kernel(const unsigned char* in_ptr0,
+                       const unsigned char* in_ptr1,
+                       unsigned char* out_ptr0)
+{
+    {
+        #pragma GCC ivdep
+        for(long i0=static_cast<long>(0L); i0<static_cast<long>(8390L); i0+=static_cast<long>(1L))
+        {
+            #pragma GCC ivdep
+            for(long i1=static_cast<long>(0L); i1<static_cast<long>(8L); i1+=static_cast<long>(1L))
+            {
+                auto tmp0 = in_ptr0[static_cast<long>(i1 + (8L*i0))];
+                auto tmp1 = in_ptr1[static_cast<long>(i1)];
+                auto tmp2 = decltype(tmp1)(-tmp1);
+                auto tmp3 = max_propagate_nan(tmp0, tmp2);
+                out_ptr0[static_cast<long>(i1 + (8L*i0))] = tmp3;
```

Comment:
would be helpful to correlate each generated line with the FX code line?

Reply:
Thanks and changed.